### ISSUE TITLE
Randomise list of users following a tag on the tag page

### ISSUE THREAD

**Issue by goobertron** [diaspora/diaspora#7634]
At the moment, there is a list of 'people tagged with #tag' in the left side-bar of each tag page.

I thought this was going to be removed during one of the updates, but it's still there.

Because it is ordered chronologically, with the first people to have followed that tag at the top, for many tags the people shown at the top of the list are all dead accounts that haven't been used for several years. See, for instance, the list one the #diaspora tag page.

This doesn't give a good impression of the health of the network.

I suggest therefore that we display the users who follow that tag in a random order, so that newer ones will (usually, at least) appear among the users at the top of the list. It will also mean that a greater range of users get noticed there, and therefore followed.

**Comment by Flaburgan** [comment]
Why not displaying it from most recent to oldest? Being tagged is a public info.

**Comment by goobertron** [comment]
If you display it from most recent to oldest, it will still display the same users at the top each time someone views that tag stream (until someone else adds the tag to their profile). Showing a random selection will mean that a different selection of users will be shown each time the page is viewed, which is, I think, good for diversity.

**Comment by SuperTux88** [comment]
Yes, I think random is better, because the newest users are probably not the most interesting, because they have less posts. And with random everybody can be at the top. Even old (still active users) can be at the top, and a new interesting user too.

**Comment by goobertron** [comment]
Of course even better would be to omit any account that hadn't posted for more than 6 months, but I'm not even going to suggest that! :-P

**Comment by stragu** [comment]
How about ordered by most recent activity?
That way, the least active would be at the bottom, and you could find both seasoned and new accounts at the top.

**Comment by SuperTux88** [comment]
Ordered by activity would be possible, but it is a much more compley query. We would need to load all posts from all people (with the tags). And the tags page is already slow in some cases. So I think random works well enough.

**Comment by stragu** [comment]
What do you mean "all post from all people"? I thought we could do:
- list accounts tagged with tag;
- associate to each account the date of its most recent post;
- order accounts by date, decreasing.

Or isn't there an easy way to fetch only the most recent post by an account?

I understand random would be faster, and in all honesty I don't really care that much, but I just wanted to understand why it would be so much more difficult.

**Comment by SuperTux88** [comment]
> Or isn't there an easy way to fetch only the most recent post by an account?

That's exactly the "problem", you need to join all posts from that author to sort them to get the latest, and that for every person with that tag It isn't unsolvable, but it's still much more complex than just randomize. And I think we should do the randomize solution first, because it solves the problem and is easy, we can always do the sort solution when randomize isn't enough.

**Comment by stragu** [comment]
Oh wow yeah that sounds resource-intensive actually!

Randomising sounds great to start with.

Thanks for explaining.

**Comment by himanshukansal** [comment]
I would like to take this up if you are not working on it @stragu.

**Comment by stragu** [comment]
@himanshukansal : I'm definitely not working on it, I am very far from having the skills :/
Thank you!

**Comment by goobertron** [comment]
@himanshukansal are you available to work on this? It would be great to give that list some love!
